error.go

```diff
@@ -20,24 +24,110 @@ import "encoding/json"
 //     }
 //
 type Error struct {
-	Desc string `json:"desc"`
-	Docs string `json:"docs"`
-	Kind string `json:"kind"`
+	Cause error   `json:"cause,omitempty"`
+	Desc  string  `json:"desc,omitempty"`
+	Docs  string  `json:"docs,omitempty"`
+	Kind  string  `json:"kind,omitempty"`
+	Stack []Stack `json:"stack,omitempty"`
```

Comment:
When we use structured logging, would it make sense to also extend that idea to errors in the sense that there would be possibility to add key-value pairs for additional contextual information? I.e. adding `Context map[string]string` here that could be used to provide information about the execution context when the error occurred.

It would help to answer better "how did the world look like when things went wrong" when the error is not fully obvious.

Reply:
Maybe it makes sense to somehow pass the same context that logger takes instead?

Reply:
I am also thinking that we haver this with the logger context already. The question is how we want to make use of it in code. I think there should not be more than one line per error handling. Everything else gets messy and I am not sure if the error callbacks introduced in the latest go versions make it better or different.